src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp

```diff
@@ -551,7 +551,7 @@ Node* ShenandoahBarrierSetC2::load_at_resolved(C2Access& access, const Type* val
     }
   }
 
-  // 3: apply keep-alive barrier if needed
+  // 3: apply keep-alive barrier for java.lang.ref.Reference if needed
```

Comment:
"For java.lang.ref.Reference"?

I think BarrierSetC2 comment if pretty generic, so we might want to match it.